integration/setup/setup.go

```diff
@@ -167,39 +172,96 @@ func WrapTestMain(c *awsclient.Client, g *framework.Guest, h *framework.Host, m
 	os.Exit(v)
 }
 
-func installAWSResource() error {
-	o := func() error {
-		// NOTE we ignore errors here because we cannot get really useful error
-		// handling done. This here should anyway only be a quick fix until we use
-		// the helm client lib. Then error handling will be better.
-		framework.HelmCmd("delete --purge aws-config-e2e")
+func installAWSOperator(h *framework.Host) error {
+	var err error
 
-		awsResourceChartValuesEnv := os.ExpandEnv(e2etemplates.ApiextensionsAWSConfigE2EChartValues)
-		d := []byte(awsResourceChartValuesEnv)
+	var values string
+	{
+		c := chartvalues.AWSOperatorConfig{
+			Guest: chartvalues.AWSOperatorConfigGuest{
+				// TODO remove this setting
+				Update: chartvalues.AWSOperatorConfigGuestUpdate{
+					Enabled: true,
+				},
+			},
+			Provider: chartvalues.AWSOperatorConfigProvider{
+				AWS: chartvalues.AWSOperatorConfigProviderAWS{
+					Encrypter: "kms",
+					Region:    env.AWSRegion(),
+				},
+			},
+			Secret: chartvalues.AWSOperatorConfigSecret{
+				AWSOperator: chartvalues.AWSOperatorConfigSecretAWSOperator{
+					IDRSAPub: env.IDRSAPub(),
+					SecretYaml: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYaml{
+						Service: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlService{
+							AWS: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWS{
+								AccessKey: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWSAccessKey{
+									ID:     env.GuestAWSAccessKeyID(),
+									Secret: env.GuestAWSAccessKeySecret(),
+									Token:  env.GuestAWSAccessKeyToken(),
+								},
+								HostAccessKey: chartvalues.AWSOperatorConfigSecretAWSOperatorSecretYamlServiceAWSAccessKey{
+									ID:     env.HostAWSAccessKeyID(),
+									Secret: env.HostAWSAccessKeySecret(),
+									Token:  env.HostAWSAccessKeyToken(),
+								},
+							},
+						},
+					},
+				},
+			},
+			RegistryPullSecret: env.RegistryPullSecret(),
+		}
 
-		err := ioutil.WriteFile(awsResourceValuesFile, d, 0644)
+		values, err = chartvalues.NewAWSOperator(c)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 
-		err = framework.HelmCmd("registry install quay.io/giantswarm/apiextensions-aws-config-e2e-chart:stable -- -n aws-config-e2e --values " + awsResourceValuesFile)
+	}
+
+	err = h.InstallBranchOperator("aws-operator", "awsconfig", values)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func installAWSConfig(h *framework.Host, vpcPeerID string) error {
+	var err error
+
+	var values string
+	{
+		c := chartvalues.APIExtensionsAWSConfigE2EConfig{
+			CommonDomain: env.CommonDomain(),
+			// TODO(PK) can this be replaced with env.ClusterID?
+			ClusterName:          env.ClusterName(),
```

Comment:
This is weird. Why it is ClusterName? Why there isn't value from `ClsuterID`. This shows getting rid of env vars is already useful.

Reply:
The env var was called cluster name for some reason, which is just wrong IMO. It should just be cluster ID. There were also TODO comments to fix that naming.

Reply:
But in the old template it was `${CLUSTER_NAME}` placeholder. And this had a different value than `env.ClusterID()`. Also `env.ClusterName() != env.ClusterID()` now.

Reply:
We can get rid of CLUSTER_NAME. It was set to CLUSTER_ID in case former was empty. And that always was the case.